modules/vmdatasource/src/xmpdatasource.cpp

```diff
@@ -268,14 +271,51 @@ void XMPDataSource::load(std::map<MetaString, std::shared_ptr<MetadataSchema> >&
     }
 }
 
+void XMPDataSource::saveStats(const std::vector< Stat* >& stats)
+{
+    statSourceCheck();
+    try
+    {
+        statSource->save(stats);
+        pushChanges();
```

Comment:
`pushChanges();` will need to be removed after we merge 'Compression' since the last one calls `pushChanges();` only once.